lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_method.rb

```diff
@@ -212,5 +213,42 @@ def self.cleanup(method_pid, threads)
       threads.each(&:exit)
     end
     private_class_method :cleanup
+
+    def self.bodies_and_line_numbers(obj, aem)
+      embeds = embedded_methods(obj.workspace, aem) << {:data => aem.data, :fqname => aem.fqname}
+      code_start = 0
+      script_info = {}
+      bodies = []
+      embeds.each do |item|
+        bodies << item[:data]
+        line_count = item[:data].lines.count
+        code_end   = code_start + line_count
+        script_info[item[:fqname]] = Range.new(code_start, code_end)
+        code_start = code_end + 1
+      end
+      return bodies, script_info
+    end
+    private_class_method :bodies_and_line_numbers
+
+    def self.embedded_methods(workspace, method_obj)
+      method_obj.embedded_methods.collect do |name|
+        method_name, klass, ns = embedded_method_name(name)
+        match_ns = workspace.overlay_method(ns, klass, method_name)
+        cls = ::MiqAeClass.find_by_fqname("#{match_ns}/#{klass}")
+        aem = ::MiqAeMethod.find_by(:class_id => cls.id, :name => method_name) if cls
+        raise MiqAeException::MethodNotFound, "Embedded method #{name} not found" unless aem
```

Comment:
Was looking specifically for this missing embedded method check and corresponding tests.  Looks good.